src/Events/Category_Colors/Repositories/Category_Color_Dropdown_Provider.php

```diff
@@ -0,0 +1,184 @@
+		 *
+		 * @since TBD
+		 *
+		 * @param array $filtered_categories The final processed categories.
+		 */
+		return (array) apply_filters( 'tec_events_category_color_dropdown_categories', $this->sort_by_priority( $filtered_categories ) );
+	}
+
+	/**
+	 * Fetches all event categories.
+	 *
+	 * @since TBD
+	 *
+	 * @return array
+	 */
+	protected function get_categories(): array {
+		$categories = get_terms(
+			[
+				'taxonomy'   => Tribe__Events__Main::TAXONOMY,
+				'hide_empty' => false,
+			]
+		);
+
+		/**
+		 * Filters the raw list of event categories before processing.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $categories The retrieved categories.
+		 */
+		return (array) apply_filters( 'tec_events_category_color_raw_categories', $categories );
+	}
+
+	/**
+	 * Retrieves category metadata (color, priority, hidden status).
+	 *
+	 * @since TBD
+	 *
+	 * @param object $category The category term object.
+	 *
+	 * @return array
+	 */
+	protected function get_category_meta( object $category ): array {
+		$meta_instance = tribe( Event_Category_Meta::class )->set_term( $category->term_id );
+		$priority      = $meta_instance->get( Meta_Keys::get_key( 'priority' ) );
+
+		$category_meta = [
+			'slug'     => $category->slug,
+			'name'     => $category->name,
+			'priority' => is_numeric( $priority ) ? (int) $priority : -1,
+			'primary'  => $meta_instance->get( Meta_Keys::get_key( 'primary' ) ),
+			'hidden'   => (bool) $meta_instance->get( Meta_Keys::get_key( 'hide_from_legend' ) ),
+		];
+
+		/**
+		 * Filters metadata of a single category.
+		 *
+		 * @since TBD
+		 *
+		 * @param array  $category_meta The metadata of the category.
+		 * @param object $category      The category term object.
+		 */
+		return (array) apply_filters( 'tec_events_category_color_category_meta', $category_meta, $category );
+	}
+
+	/**
+	 * Filters categories based on their primary color and visibility settings.
+	 *
+	 * @since TBD
+	 *
+	 * @param array $categories The list of categories.
+	 *
+	 * @return array
+	 */
+	protected function filter_categories( array $categories ): array {
+		$show_hidden_categories = tribe_get_option( 'category-color-show-hidden-categories', false );
+
+		$filtered = array_values(
+			array_filter(
+				$categories,
+				fn( $category ) => ! empty( $category['primary'] ) && ( $show_hidden_categories || ! $category['hidden'] )
+			)
+		);
+
+		/**
+		 * Filters the categories after visibility filtering.
+		 *
+		 * @since TBD
+		 *
+		 * @param array $filtered The filtered categories list.
+		 */
+		return (array) apply_filters( 'tec_events_category_color_filtered_categories', $filtered );
+	}
+
+	/**
+	 * Sorts categories by priority (highest first).
+	 *
+	 * @since TBD
+	 *
+	 * @param array $categories The list of categories.
+	 *
+	 * @return array
+	 */
+	protected function sort_by_priority( array $categories ): array {
+		usort( $categories, fn( $a, $b ) => $b['priority'] <=> $a['priority'] );
```

Comment:
We should have some isset check for the `priority` key as the values are coming from filtered data and we can't trust that.